kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveStores.java

```diff
@@ -0,0 +1,141 @@
+package dev.responsive.kafka.api;
+
+import java.time.Duration;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.kstream.JoinWindows;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.StreamJoined;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.TimestampedKeyValueStore;
+import org.apache.kafka.streams.state.TimestampedWindowStore;
+import org.apache.kafka.streams.state.WindowBytesStoreSupplier;
+import org.apache.kafka.streams.state.WindowStore;
+
+/**
+ * A factory for creating Kafka Streams state stores on top of a Responsive storage backend.
+ * Use these when building your {@link org.apache.kafka.streams.Topology} to easily swap in
+ * Responsive stores wherever state is used.
+ *
+ * See {@link org.apache.kafka.streams.state.Stores} for instructions on how to plug in custom
+ * state stores and configure them.
+ */
+public final class ResponsiveStores {
+
+  /**
+   * Create a {@link StoreBuilder} that can be used to build a Responsive
+   * {@link KeyValueStore} and connect it via the Processor API. If using the DSL, use
+   * {@link #materialized(KeyValueStoreOptions)} instead.
+   *
+   * @param options the {@link KeyValueStoreOptions} to use
+   * @return a store builder that can be used to build a key-value store with the given options
+   *         that uses Responsive's storage for its backend
+   */
+  public static <K, V> StoreBuilder<KeyValueStore<K, V>> keyValueStoreBuilder(
```

Comment:
what will we do if someone tries to set something on one of these builders that conflicts with the passed in option? (e.g. `withLoggingEnabled` with some topic config that conflicts like setting compact when `truncateChangelongs` is true)

Reply:
Ah I see from pt3 we're throwing an exception. Very nice.